### DISCUSSION TITLE
not working

### DISCUSSION THREAD

**Discussion by Lostinaa** [taj-ny/kwin-effects-forceblur#130] [General]
this isnt working can u please help me ive installed and built and done everything u said but its not working

**Comment by taj-ny** [comment]
What exactly do you mean by "isnt working"? Is it not showing up in desktop effects? What distribution and Plasma version are you using?

**Reply by Lostinaa** [reply]
i use arch kde 6.2 but its working now thank you bro.. it worked when i turned of window opacity affects blur

**Reply by Lostinaa** [reply]
but one other simple problem i had is when i turn of the the blur(default) form desktop effects the other windows also uses the force blur effect even though i didnt specify their class... but when i turn on the blur it uses the its own previous blur and its okay

**Reply by Lostinaa** [reply]
now it stopped again it was working for a few days then even the whole setting in desktop effects disappeared

**Reply by taj-ny** [reply]
Rebuild the effect, you need to do that every time Plasma is updated.

**Reply by Lostinaa** [reply]
i did that still not working

**Reply by taj-ny** [reply]
Did you remove the *build* directory?

**Reply by Lostinaa** [reply]
yeah i even removed the whole git directory and cloned it again

**Reply by taj-ny** [reply]
Send the build log.

**Reply by Lostinaa** [reply]
The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Installing in the same prefix as Qt, adopting their path scheme.
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Performing Test _OFFT_IS_64BIT
-- Performing Test _OFFT_IS_64BIT - Success
-- Performing Test HAVE_DATE_TIME
-- Performing Test HAVE_DATE_TIME - Success
-- Found OpenGL: /usr/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found XKB: /usr/lib/libxkbcommon.so (found suitable version "1.7.0", minimum required is "0.5.0")
-- Found WrapVulkanHeaders: /usr/include
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found KF6Config: /usr/lib/cmake/KF6Config/KF6ConfigConfig.cmake (found version "6.9.0")
-- Found KF6ConfigWidgets: /usr/lib/cmake/KF6ConfigWidgets/KF6ConfigWidgetsConfig.cmake (found version "6.9.0")
-- Found KF6CoreAddons: /usr/lib/cmake/KF6CoreAddons/KF6CoreAddonsConfig.cmake (found version "6.9.0")
-- Found KF6Crash: /usr/lib/cmake/KF6Crash/KF6CrashConfig.cmake (found version "6.9.0")
-- Found KF6GlobalAccel: /usr/lib/cmake/KF6GlobalAccel/KF6GlobalAccelConfig.cmake (found version "6.9.0")
-- Found Gettext: /usr/bin/msgmerge (found version "0.22.5")
-- Found KF6I18n: /usr/lib/cmake/KF6I18n/KF6I18nConfig.cmake (found version "6.9.0")
-- Found X11: /usr/include
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so
-- Looking for XOpenDisplay in /usr/lib/libX11.so;/usr/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found KF6KIO: /usr/lib/cmake/KF6KIO/KF6KIOConfig.cmake (found version "6.9.0")
-- Found KF6Service: /usr/lib/cmake/KF6Service/KF6ServiceConfig.cmake (found version "6.9.0")
-- Found KF6Notifications: /usr/lib/cmake/KF6Notifications/KF6NotificationsConfig.cmake (found version "6.9.0")
-- Found KF6WidgetsAddons: /usr/lib/cmake/KF6WidgetsAddons/KF6WidgetsAddonsConfig.cmake (found version "6.9.0")
-- Found KF6WindowSystem: /usr/lib/cmake/KF6WindowSystem/KF6WindowSystemConfig.cmake (found version "6.9.0")
-- Found KF6GuiAddons: /usr/lib/cmake/KF6GuiAddons/KF6GuiAddonsConfig.cmake (found version "6.9.0")
-- Found KF6KCMUtils: /usr/lib/cmake/KF6KCMUtils/KF6KCMUtilsConfig.cmake (found version "6.9.0")
-- Found KF6: success (found suitable version "6.9.0", minimum required is "5.240.0") found components: Config ConfigWidgets CoreAddons Crash GlobalAccel I18n KIO Service Notifications Service WidgetsAddons WindowSystem GuiAddons KCMUtils
-- Found epoxy: /usr/lib/libepoxy.so (found version "1.5.10")
-- Found XCB_XCB: /usr/lib/libxcb.so (found version "1.17.0")
-- Found XCB: /usr/lib/libxcb.so (found version "1.17.0") found components: XCB
-- Found Wayland_Server: /usr/lib/libwayland-server.so (found version "1.23.1")
-- Found Wayland: /usr/lib/libwayland-server.so (found version "1.23.1") found components: Server
CMake Warning at CMakeLists.txt:84 (find_package):
  By not providing "FindKDecoration3.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "KDecoration3", but CMake did not find one.

  Could not find a package configuration file provided by "KDecoration3" with
  any of the following names:

    KDecoration3Config.cmake
    kdecoration3-config.cmake

  Add the installation prefix of "KDecoration3" to CMAKE_PREFIX_PATH or set
  "KDecoration3_DIR" to a directory containing one of the above files.  If
  "KDecoration3" provides a separate development package or SDK, be sure it
  has been installed.


-- The following OPTIONAL packages have been found:

 * OpenGL
 * Qt6UiTools (required version >= 6.6.0)
 * Qt6WidgetsTools (required version >= 6.8.1)
 * Qt6Xml (required version >= 6.6.0)
 * Qt6Network (required version >= 6.6.0)
 * Qt6DBusTools (required version >= 6.8.1)
 * Qt6QmlTools (required version >= 6.8.1)
 * Qt6CoreTools (required version >= 6.8.1)
 * XKB (required version >= 0.5.0), XKB API common to servers and clients., <http://xkbcommon.org>
 * WrapVulkanHeaders
 * Qt6GuiTools (required version >= 6.8.1)
 * KDecoration2

-- The following REQUIRED packages have been found:

 * ECM
 * Qt6 (required version >= 6.6.0)
 * KF6Crash (required version >= 5.240.0)
 * KF6GlobalAccel (required version >= 5.240.0)
 * Gettext
 * KF6I18n (required version >= 5.240.0)
 * KF6KIO (required version >= 5.240.0)
 * KF6Notifications (required version >= 5.240.0)
 * KF6Service (required version >= 5.240.0)
 * KF6WidgetsAddons (required version >= 5.240.0)
 * KF6GuiAddons (required version >= 5.240.0)
 * KF6ConfigWidgets (required version >= 6.9.0)
 * KF6KCMUtils (required version >= 5.240.0)
 * KF6 (required version >= 5.240.0)
 * XCB, X protocol C-language Binding, <https://xcb.freedesktop.org/>
 * X11
 * epoxy
 * KWin
 * Qt6Gui (required version >= 6.7.0)
 * KWinDBusInterface

-- The following OPTIONAL packages have not been found:

 * KDecoration3

-- Configuring done (2.2s)
-- Generating done (0.0s)
-- Build files have been written to: /home/dopy/kwin-effects-forceblur/build

**Reply by Lostinaa** [reply]
[  0%] Built target forceblur_autogen_timestamp_deps
[  0%] Built target kwin_better_blur_config_autogen_timestamp_deps
[  9%] Automatic MOC for target kwin_better_blur_config
[  9%] Automatic MOC for target forceblur
AutoMoc warning
---------------
"SRC:/src/main.cpp"
includes the moc file "main.moc", but does not contain a Q_OBJECT, Q_GADGET, Q_NAMESPACE, Q_NAMESPACE_EXPORT, Q_GADGET_EXPORT, Q_ENUM_NS, K_PLUGIN_FACTORY, K_PLUGIN_CLASS, K_PLUGIN_FACTORY_WITH_JSON or K_PLUGIN_CLASS_WITH_JSON macro.

[  9%] Built target kwin_better_blur_config_autogen
[ 13%] Generating ui_blur_config.h
[ 18%] Generating blurconfig.h, blurconfig.cpp
[ 22%] Generating kwineffects_interface.cpp, kwineffects_interface.h
/home/dopy/kwin-effects-forceblur/src/kcm/blur_config.ui: Warning: The name 'horizontalSpacer_2' (QSpacerItem) is already in use, defaulting to 'horizontalSpacer_21'.
[ 27%] Generating moc_kwineffects_interface.cpp
[ 27%] Built target forceblur_autogen
[ 36%] Automatic RCC for blur.qrc
[ 36%] Generating blurconfig.h, blurconfig.cpp
[ 40%] Building CXX object src/CMakeFiles/forceblur.dir/blurconfig.cpp.o
[ 50%] Building CXX object src/CMakeFiles/forceblur.dir/main.cpp.o
[ 50%] Building CXX object src/CMakeFiles/forceblur.dir/forceblur_autogen/mocs_compilation.cpp.o
[ 54%] Building CXX object src/CMakeFiles/forceblur.dir/settings.cpp.o
[ 59%] Building CXX object src/CMakeFiles/forceblur.dir/blur.cpp.o
[ 63%] Building CXX object src/CMakeFiles/forceblur.dir/forceblur_autogen/EWIEGA46WW/qrc_blur.cpp.o
[ 72%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/blurconfig.cpp.o
[ 72%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/blur_config.cpp.o
[ 77%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/kwin_better_blur_config_autogen/mocs_compilation.cpp.o
[ 81%] Building CXX object src/kcm/CMakeFiles/kwin_better_blur_config.dir/kwineffects_interface.cpp.o
/home/dopy/kwin-effects-forceblur/src/blur.cpp: In member function ‘virtual void KWin::BlurEffect::reconfigure(KWin::Effect::ReconfigureFlags)’:
/home/dopy/kwin-effects-forceblur/src/blur.cpp:235:47: warning: unused parameter ‘flags’ [-Wunused-parameter]
  235 | void BlurEffect::reconfigure(ReconfigureFlags flags)
      |                              ~~~~~~~~~~~~~~~~~^~~~~
/home/dopy/kwin-effects-forceblur/src/blur.cpp: In member function ‘void KWin::BlurEffect::blur(KWin::BlurRenderData&, const KWin::RenderTarget&, const KWin::RenderViewport&, KWin::EffectWindow*, int, const QRegion&, KWin::WindowPaintData&)’:
/home/dopy/kwin-effects-forceblur/src/blur.cpp:749:138: warning: unused parameter ‘mask’ [-Wunused-parameter]
  749 | void BlurEffect::blur(BlurRenderData &renderInfo, const RenderTarget &renderTarget, const RenderViewport &viewport, EffectWindow *w, int mask, const QRegion &region, WindowPaintData &data)
      |                                                                                                                                      ~~~~^~~~
[ 86%] Linking CXX shared module kwin_better_blur_config.so
[ 90%] Built target kwin_better_blur_config
[ 95%] Linking CXX shared module forceblur.so
[100%] Built target forceblur

**Reply by taj-ny** [reply]
And did you run ``sudo make install``?

**Reply by Lostinaa** [reply]
i did now inside the build directory and it worked.. i figured last time i didnt do that and it was working and i thought it'd do the same now

**Reply by Lostinaa** [reply]
thanks for ur help tho i appreciate it
